api/resolvers/domains.py

```diff
@@ -2,59 +2,148 @@
 from sqlalchemy.orm import load_only
 
 from app import app
+from db import db
+
+from functions.auth_wrappers import require_token
+from functions.auth_functions import is_super_admin, is_user_read
+
+from models import Domains, Organizations
+
+from schemas.domain import Domain
+
+
+@require_token
+def resolve_domain(self: Domain, info, **kwargs):
+    """
+    This function is to resolve the domain query which takes in a url as a URL
+    scalar and checks against the user roles passed in by the Authorization
+    header, and return filtered results depending on the users access level
+    :param self: Domain SQLAlchemyObject type defined in the schemas directory
+    :param info: Request information sent to the sever from a client
+    :param kwargs: Field arguments (i.e. url), and user_roles
+    :return: Filtered Domain SQLAlchemyObject Type
+    """
+    # Get Information passed in via kwargs
+    url = kwargs.get('url')
+    user_role = kwargs.get('user_roles')
+
+    # Generate list of org's the user has access to
+    org_id_list = []
+    for role in user_role:
+        org_id_list.append(role["org_id"])
+
+    # Get initial Domain Query Object
+    query = Domain.get_query(info)
+
+    # Check to see if the user is a super admin, if true return all information
+    if is_super_admin(user_role=user_role):
+        query_rtn = query.filter(
+            Domains.domain == url
+        ).all()
+        # If url cannot be found, raise error
+        if not len(query_rtn):
+            raise GraphQLError("Error, domain does not exist")
+    # If user fails super user test
+    else:
+        # Get org id that is related to the domain
+        org_orm = db.session.query(Organizations).filter(
+            Organizations.id == Domains.organization_id
+        ).filter(
+            Domains.domain == url
+        ).first()
+
+        # If org cannot be found
+        if not org_orm:
+            raise GraphQLError("Error, domain does not exist")
+        org_id = org_orm.id
+
+        # Check if user has read access or higher to the requested organization
+        if is_user_read(user_role, org_id):
+            query_rtn = query.filter(
+                Domains.domain == url
+            ).all()
+        else:
+            raise GraphQLError("Error, you do not have permission to view that domain")
```

Comment:
I think the `Error,` here is redundant?  Won't it say: Error: Error, you do not have permission.... ???

Reply:
No it will not have duplicated 'Error', it is however located in the `errors` list object